src/entities/trade.rs

```diff
@@ -500,6 +504,39 @@ mod tests {
     mod new {
         use super::*;
 
+        #[test]
+        fn calculates_input_and_output_amounts_correctly() {
+            let trade = Trade::new(
+                Route::new(vec![PAIR_0_1.clone()], TOKEN0.clone(), TOKEN1.clone()),
+                CurrencyAmount::from_raw_amount(TOKEN0.clone(), 100).unwrap(),
+                TradeType::ExactInput,
+            )
+            .unwrap();
+            assert_eq!(trade.input_amount.quotient(), BigInt::from(100));
+            assert!(trade.output_amount.quotient() < BigInt::from(100));
+        }
+
+        #[test]
+        fn calculates_price_impact_correctly() {
+            let trade = Trade::new(
+                Route::new(vec![PAIR_0_1.clone()], TOKEN0.clone(), TOKEN1.clone()),
+                CurrencyAmount::from_raw_amount(TOKEN0.clone(), 100).unwrap(),
+                TradeType::ExactInput,
+            )
+            .unwrap();
+            assert!(trade.price_impact > Percent::new(0, 100));
+        }
+
+        #[test]
+        fn fails_with_insufficient_reserves() {
+            let result = Trade::new(
+                Route::new(vec![EMPTY_PAIR_0_1.clone()], TOKEN0.clone(), TOKEN1.clone()),
+                CurrencyAmount::from_raw_amount(TOKEN0.clone(), 100).unwrap(),
+                TradeType::ExactInput,
+            );
+            assert!(result.is_err());
+        }
```

Comment:
I don't know where these came from. Follow the tests in v2-sdk.
https://github.com/Uniswap/v2-sdk/blob/main/src/entities/trade.test.ts